### PR TITLE
Custom components

### PR DESCRIPTION
**Pull request by michaelmcauley**
Adds support for custom footer & empty state components. Also adds debugMode flag to help diagnose issues in fixtable-core, and fixes a bug with content rows scrolling behind footer in some cases.

### CODE REVIEW

**Comment by hornetnz** [comment]
Definitely a step in the right direction!

**Comment by michaelmcauley** [comment]
@hornetnz Yeah! When I have time I want to add support for many more custom components so everything can be extended or overridden as needed.

File: addon/components/fixtable-footer.js [hunk 1]
```diff
@@ -3,7 +3,6 @@ import layout from '../templates/components/fixtable-footer';
 
 export default Ember.Component.extend({
   layout,
-  classNames: ['fixtable-footer'],
```
Comment (rbrcurtis) [inline addon/components/fixtable-footer.js]: is this going to be a breaking change?

Comment (michaelmcauley) [reply addon/components/fixtable-footer.js]: This class moves from being directly on the component to being applied to a new div that wraps the component, so it shouldn't be a problem, and hasn't been an issue in testing this linked against our Ember engine.

Comment (rbrcurtis) [reply addon/components/fixtable-footer.js]: Cool, sounds good.

**Pull request merged**